RFCs/0014-ToJsonString.md

```diff
@@ -0,0 +1,113 @@
+    }
+}
+```
+
+With the new `Fn::ToJsonString` intrinsic function we can simplify to the following:
+
+## JSON
+```json
+"MyDashboard": {
+    "Type": "AWS::CloudWatch::Dashboard",
+    "Properties": {
+        "DashboardBody": {
+            "Fn::ToJsonString": {
+                "start": "-PT6H",
+                "periodOverride": "inherit",
+                "widgets": [
+                    {  
+                        "type": "text",
+                        "x": 0,
+                        "y": 7,
+                        "width": 3,
+                        "height": 3,
+                        "properties": { "markdown": "Hello world" }
+                    }
+                ]
+            }
+        }
+    }
+}
+```
+
+## YAML
+```yaml
+MyDashboard:
+    Type: AWS::CloudWatch::Dashboard
+    Properties:
+        DashboardBody:
+            Fn::ToJsonString:
+                start: "-PT6H"
+                periodOverride: inherit
+                widgets:
+                    - type: text
+                    x: 0
+                    y: 7
+                    width: 3
+                    height: 3
+                    properties:
+                        markdown: Hello world
+```
+
+### Short Form Syntax
+```yaml
+MyDashboard:
+    Type: AWS::CloudWatch::Dashboard
+    Properties:
+        DashboardBody:
+            !ToJsonString:
+                start: "-PT6H"
+                periodOverride: inherit
+                widgets:
+                    - type: text
+                    x: 0
+                    y: 7
+                    width: 3
+                    height: 3
+                    properties:
+                        markdown: Hello world
+```
+
+# Details
+
+`Fn::ToJsonString` is an intrinsic function that takes in a template block as input and converts it into an escaped JSON string.
+
+* It will be restricted to only be used as the value of string-type resource properties.
+* Intrinsic functions (e.g. `Fn::If`, `Ref`) or pseudo parameters (e.g. `AWS::NoValue`) can be used within the input template block. The input template block will be processed by the intrinsic functions and pseudo parameters before it is converted to a string (i.e. the resolved value of the intrinsic function is converted to the output string, and not the intrinsic function itself).
+* Conversion will always retain the same order of key-value pairs such that the converted strings of the same input template block are guaranteed to not change.
+* The following pseudo parameters will be supported:
+    * `AWS::AccountId`, `AWS::Region`, `AWS::Partition`, `AWS::NoValue`
+
+# Limitation
+
+* YAML is a superset of JSON, so there are a few features in YAML that can not be converted to JSON due to their differences. For example, comments are supported in YAML but not JSON. This can potentially be a confusing developer experience if the developer is not aware of these limitations and have different expectations.
```

Comment:
Can we be explicit about what happens with unsupported yaml features?
eg. I would like it to work even if there are comments inside the nested yaml.

Use case:

```yaml
MyDashboard:
    Type: AWS::CloudWatch::Dashboard
    Properties:
        DashboardBody:
            !ToJsonString:
                start: "-PT6H"  # 6 hours ago
[...]
```

Reply:
Yes, the description here can be more clear. Your example with using comments in the yaml block should still be considered valid, but the json string it converts into will lose that comment (as the json data format does not support comments).